RecoLocalCalo/HGCalRecProducers/plugins/alpaka/HGCalLayerClustersSoAAlgoWrapper.dev.cc

```diff
@@ -0,0 +1,228 @@
+
+        if (outputs_service[cluster_index].total_weight_log() > 0.f) {
+          float inv_tot_weight = 1.f / outputs_service[cluster_index].total_weight_log();
+          outputs[cluster_index].x() *= inv_tot_weight;
+          outputs[cluster_index].y() *= inv_tot_weight;
+        } else {
+          outputs[cluster_index].x() = input_rechits_soa[max_energy_index].dim1();
+          outputs[cluster_index].y() = input_rechits_soa[max_energy_index].dim2();
+        }
+        outputs[cluster_index].z() = input_rechits_soa[max_energy_index].dim3();
+      }  // uniform_elements
+    }    // operator()
+  };
+
+  void HGCalLayerClustersSoAAlgoWrapper::run(Queue& queue,
+                                             const unsigned int size,
+                                             float thresholdW0,
+                                             float positionDeltaRho2,
+                                             const HGCalSoARecHitsDeviceCollection::ConstView input_rechits_soa,
+                                             const HGCalSoARecHitsExtraDeviceCollection::ConstView input_clusters_soa,
+                                             HGCalSoAClustersDeviceCollection::View outputs,
+                                             HGCalSoAClustersExtraDeviceCollection::View outputs_service) const {
+    auto x = cms::alpakatools::make_device_view<float>(alpaka::getDev(queue), outputs.x(), size);
+    alpaka::memset(queue, x, 0x0);
+    auto y = cms::alpakatools::make_device_view<float>(alpaka::getDev(queue), outputs.y(), size);
+    alpaka::memset(queue, y, 0x0);
+    auto z = cms::alpakatools::make_device_view<float>(alpaka::getDev(queue), outputs.z(), size);
+    alpaka::memset(queue, z, 0x0);
+    auto seed = cms::alpakatools::make_device_view<int>(alpaka::getDev(queue), outputs.seed(), size);
+    alpaka::memset(queue, seed, 0x0);
+    auto energy = cms::alpakatools::make_device_view<float>(alpaka::getDev(queue), outputs.energy(), size);
+    alpaka::memset(queue, energy, 0x0);
+    auto cells = cms::alpakatools::make_device_view<int>(alpaka::getDev(queue), outputs.cells(), size);
+    alpaka::memset(queue, cells, 0x0);
+    auto total_weight =
+        cms::alpakatools::make_device_view<float>(alpaka::getDev(queue), outputs_service.total_weight(), size);
+    alpaka::memset(queue, total_weight, 0x0);
+    auto total_weight_log =
+        cms::alpakatools::make_device_view<float>(alpaka::getDev(queue), outputs_service.total_weight_log(), size);
+    alpaka::memset(queue, total_weight_log, 0x0);
+    auto maxEnergyValue =
+        cms::alpakatools::make_device_view<float>(alpaka::getDev(queue), outputs_service.maxEnergyValue(), size);
+    alpaka::memset(queue, maxEnergyValue, 0x0);
+    auto maxEnergyIndex =
+        cms::alpakatools::make_device_view<int>(alpaka::getDev(queue), outputs_service.maxEnergyIndex(), size);
+    alpaka::memset(queue, maxEnergyIndex, 0xff);
```

Comment:
Could you add a comment that this fills the memory buffer with `-1` (or `kInvalidCluster` if you reuse the same constant here) ?